source/features/ci-link.tsx

```diff
@@ -5,31 +5,51 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 import fetchDom from '../helpers/fetch-dom';
-import {buildRepoURL} from '../github-helpers';
+import {buildRepoURL, getConversationNumber} from '../github-helpers';
 
 // Look for the CI icon in the latest 2 days of commits #2990
-const getIcon = onetime(async () => fetchDom(
+const getRepoIcon = onetime(async () => fetchDom(
 	buildRepoURL('commits'), [
 		'.commit-group:nth-of-type(-n+2) .commit-build-statuses', // Pre "Repository refresh" layout
 		'.TimelineItem--condensed:nth-of-type(-n+2) .commit-build-statuses'
 	].join()
 ));
 
-async function init(): Promise<false | void> {
-	const icon = await getIcon() as HTMLElement | undefined;
+const getPRIcon = onetime(async () => fetchDom(
+	buildRepoURL('pull', getConversationNumber()!, 'commits'), '.js-commits-list-item:last-of-type .commit-build-statuses'
+));
+
+async function initRepo(): Promise<false | void> {
+	const icon = await getRepoIcon() as HTMLElement | undefined;
 	if (!icon) {
 		return false;
 	}
 
 	icon.classList.add('rgh-ci-link');
-	if (onetime.callCount(getIcon) > 1) {
+	if (onetime.callCount(getRepoIcon) > 1) {
 		icon.style.animation = 'none';
 	}
 
 	// Append to title (aware of forks and private repos)
 	select('[itemprop="name"]')!.parentElement!.append(icon);
 }
 
+async function initPR(): Promise<false | void> {
+	const icon = await getPRIcon() as HTMLElement | undefined;
+	if (!icon) {
+		return false;
+	}
+
+	icon.classList.add('rgh-ci-link');
+	if (onetime.callCount(getPRIcon) > 1) {
+		icon.style.animation = 'none';
+	}
+
+	// Append to PR title and floating header
+	select('.gh-header-title')!.append(icon);
+	select('.gh-header-number')!.after(icon.cloneNode(true));
```

Comment:
Do the icons stay up there even after new comments, label changes, title changes?

Reply:
Nope... Guess we need selector-observer

Reply:
```js
	init() {
		observeElement(select('#partial-discussion-header')!.parentElement!, init);
	}
```


This should be enough no?

Reply:
Hopefully 👆